pymatgen/io/openmm/utils.py

```diff
@@ -360,111 +444,84 @@ def parameterize_system(
-            box_vectors = list(np.array(box[3:6]) - np.array(box[0:3])) * angstrom
-            openff_topology.box_vectors = box_vectors
-            system = openff_forcefield.create_openmm_system(
-                openff_topology,
-                charge_from_molecules=charged_openff_mols,
-                allow_nonintegral_charges=True,
-            )
-            return system
+        ff_name = force_field.lower()
+        charged_off_mols = assign_charges_to_mols(
+            smile_strings=smile_strings,
+            partial_charges=partial_charges,
+            partial_charge_scaling=partial_charge_scaling,
+            partial_charge_method=partial_charge_method,
+        )
+        template = assign_small_molecule_ff(molecules=charged_off_mols, forcefield_name=ff_name)
+        forcefield_omm.registerTemplateGenerator(template.generator)
+
     else:
-        # TODO: Make decisions for user about ff name
-        # TODO: Dict instead of list of tuples
-        # TODO: Make periodic
-        # TODO: figure out how to add partial charges
-        small_ffs = [
-            "smirnoff99Frosst-1.0.2",
-            "smirnoff99Frosst-1.0.0",
-            "smirnoff99Frosst-1.1.0",
-            "smirnoff99Frosst-1.0.4",
-            "smirnoff99Frosst-1.0.8",
-            "smirnoff99Frosst-1.0.6",
-            "smirnoff99Frosst-1.0.3",
-            "smirnoff99Frosst-1.0.1",
-            "smirnoff99Frosst-1.0.5",
-            "smirnoff99Frosst-1.0.9",
-            "smirnoff99Frosst-1.0.7",
-            "openff-1.0.1",
-            "openff-1.1.1",
-            "openff-1.0.0-RC1",
-            "openff-1.2.0",
-            "openff-1.3.0",
-            "openff-2.0.0-rc.2",
-            "openff-2.0.0",
-            "openff-1.1.0",
-            "openff-1.0.0",
-            "openff-1.0.0-RC2",
-            "openff-1.3.1",
-            "openff-1.2.1",
-            "openff-1.3.1-alpha.1",
-            "openff-2.0.0-rc.1",
-            "gaff-1.4",
-            "gaff-1.8",
-            "gaff-1.81",
-            "gaff-2.1",
-            "gaff-2.11",
-        ]
         small_molecules = {}
-        large_or_water = {}
+        biopolymer_or_water = []
+        # iterate through each smile, if no forcefielded provided use Sage
         # iterate through each molecule and forcefield input as list
-        for smile, ff_name in force_field.items():
-            openff_mol = openff.toolkit.topology.Molecule.from_smiles(smile)
-            # Assign mols and forcefield as small molecule vs AMBER or
-            # CHARMM
-            if ff_name.lower() in small_ffs:
-                small_molecules[openff_mol] = ff_name.lower()
+        # Add charges to the molecule if provided
+        charged_mols = assign_charges_to_mols(
+            smile_strings=smile_strings,
+            partial_charges=partial_charges,
+            partial_charge_method=partial_charge_method,
+            partial_charge_scaling=partial_charge_scaling,
+        )
```

Comment:
can we pull this out front and call it once for small molecules and biopolymer_or_water?